src/main/java/org/spongepowered/common/event/tracking/phase/packet/PacketPhaseUtil.java

```diff
@@ -165,8 +165,22 @@ public static boolean allTransactionsInvalid(final List<SlotTransaction> slotTra
     @SuppressWarnings({"rawtypes", "unchecked"})
     public static void onProcessPacket(final Packet packetIn, final INetHandler netHandler) {
         if (netHandler instanceof NetHandlerPlayServer) {
+            EntityPlayerMP packetPlayer = ((NetHandlerPlayServer) netHandler).player;
+            if (!packetPlayer.isEntityAlive()) {
+                // Don't process packets from a player when they are dead.
+                if (!(packetIn instanceof CPacketClientStatus)) {
+                    return;
+                }
+
+                CPacketClientStatus clientStatusPacket = (CPacketClientStatus) packetIn;
```

Comment:
You could likely collapse it down to one if, to lower the diff.

Reply:
I've collapsed it slightly, I've left the `isEntityAlive` if statement as it easier to read if it's separate and it allows additional logic to be added in future if needed